### DISCUSSION TITLE
Inference on all Test Data Split at once

### DISCUSSION THREAD

**Discussion by JPVercosa** [ultralytics/ultralytics#1428] [Q&A] [answered]
What is the best approach to inference on all images in the test data split using a custom-trained YOLOv8 model?

Let me clarify that I have already trained my custom YOLOv8 model and I know how to use the "mode=predict" functionality to perform inference on individual images. However, I am looking for a way to perform inference on all of the images in my test data split at once, rather than one at a time. Ideally, I would like to automate this process as much as possible.

Furthermore, my test images are labeled, and I would like to be able to evaluate the performance of my model on this test data using metrics such as precision, recall, and F1-score. Are there any tools or libraries that can help me perform this evaluation? Is there a recommended workflow for evaluating the performance of a YOLOv8 model on labeled **test** dataset?

Any guidance or advice on how to approach these challenges would be greatly appreciated. Thank you!

**Comment by JPVercosa** [comment, marked as answer]
I've found that there is a parameter called split: 

![image](https://user-images.githubusercontent.com/27090431/225125821-af3b768f-767c-41cd-83f4-a8a24ad8a4ba.png)

When using validation, you just need to add it, and the problem is solved.

**Reply by glenn-jocher** [reply]
@JPVercosa great to hear that you've found the `split` parameter useful! Indeed, for running inference on your entire test dataset, you can use the `predict` mode with the `split` parameter set to 'test'. This will automate the process and apply your custom-trained YOLOv8 model to all images in the specified test split.

To evaluate the performance of your model on the labeled test data, you can use the `val` mode with the `split` parameter set to 'test'. This will provide you with metrics such as precision, recall, and F1-score, which are essential for understanding your model's performance on the test dataset.

Remember, the evaluation metrics are only as good as the quality of your labeled data, so ensure that your test set is well-annotated. For more detailed information on these modes and how to use them, please refer to our documentation at https://docs.ultralytics.com. Keep up the good work! 🚀
